CarRecognition/Source Files/Common/Services/CarClassificationService.swift

```diff
@@ -69,16 +81,62 @@ internal final class CarClassificationService {
         self.state = state
     }
     
-    private func handleDetection(request: VNRequest, error: Error?) {
-        guard
-            let results = request.results,
-            let classifications = results as? [VNClassificationObservation]
-        else {
-            return
+    private func performClassification(on pixelBuffer: CVPixelBuffer) {
+        guard state == .running else { return }
+        if type == .detection {
+            guard isReadyForNextFrame else { return }
+        }
+        currentBuffer = pixelBuffer
+        let orientation = CGImagePropertyOrientation.right
+        let handler = VNImageRequestHandler(cvPixelBuffer: pixelBuffer, orientation: orientation, options: [:])
+        DispatchQueue.global(qos: .userInitiated).async {
+            var request: VNCoreMLRequest
+            switch self.type {
+            case .detection:
+                request = self.detectionRequest
+            case .recognition:
+                request = self.recognitionRequest
+            }
+            try? handler.perform([request])
+        }
+    }
+    
+    private func handleResults(request: VNRequest, error: Error?, for type: ClassificationType) {
+        guard let results = request.results,
```

Comment:
We usually writing it this way when dividing into multiple lines :)
![zrzut ekranu 2018-10-11 o 17 10 28](https://user-images.githubusercontent.com/5294256/46814284-a272c400-cd78-11e8-8711-fe98b5741726.png)

Reply:
Oh, actually, last line should be one tabulation to the left, didn't notice :)

Reply:
Hm, I don't think that this is according to the Swift Style Guidelines.